ImputationPipeline/AggregatePRSResults.wdl

```diff
@@ -57,27 +57,27 @@ task AggregateResults {
     library(magrittr)
     library(ggplot2)
 
-    results <- c("~{sep='","' results}") %>% map(read_csv, col_types=cols(.default = 'c')) %>% reduce(bind_rows)
+    results <- c("~{sep='","' results}") %>% map(read_csv, col_types=cols(is_control='l', .default='c')) %>% reduce(bind_rows)
     
-    num_batches <- lapply(results, function(row) {length(unique(row))})$batch_id
+    batch_ids <- results %>% pull(batch_id) %>% unique()
 
-    if (num_batches != 1) {
-      stop(paste0("There are ", num_batches, " batch IDs in the input tables, however, only 1 is expected."))
+    if (length(batch_ids) != 1) {
+      stop(paste0("There are ", length(batch_ids), " batch IDs in the input tables, however, only 1 is expected."))
     }
 
-    batch_id <- results$batch_id[1]
+    batch_id <- batch_ids[1]
```

Comment:
you don't actually need to select the particular element here.  In R a single value is actually vector of length 1.  So, since batch_ids is guaranteed to be length 1 a this point in the code, `batch_ids`, `batch_ids[1]`, and `batch_ids[[1]]` are all the same.

Reply:
Wizardry!